test/cli/test/artifacts.test.ts

```diff
@@ -0,0 +1,407 @@
+              "column": 9,
+              "file": "<root>/basic.test.ts",
+              "line": 7,
+            },
+            "type": "2",
+          },
+          {
+            "attachments": [
+              {
+                "contentType": "text/javascript",
+                "path": "<root>/.vitest-attachments/<hash>.js",
+              },
+            ],
+            "location": {
+              "column": 9,
+              "file": "<root>/basic.test.ts",
+              "line": 8,
+            },
+            "type": "3",
+          },
+          {
+            "attachments": [
+              {
+                "contentType": "text/javascript",
+                "path": "<root>/.vitest-attachments/<hash>.js",
+              },
+            ],
+            "location": {
+              "column": 9,
+              "file": "<root>/basic.test.ts",
+              "line": 9,
+            },
+            "type": "4",
+          },
+          {
+            "location": {
+              "column": 9,
+              "file": "<root>/basic.test.ts",
+              "line": 10,
+            },
+            "type": "external",
+          },
+          {
+            "attachments": [
+              {
+                "body": "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/",
+              },
+            ],
+            "location": {
+              "column": 9,
+              "file": "<root>/basic.test.ts",
+              "line": 11,
+            },
+            "type": "with base64 body",
+          },
+          {
+            "attachments": [
+              {
+                "body": "AAECAwQFBgcICQoLDA0ODxAREhMUFRYXGBkaGxwdHh8gISIjJCUmJygpKissLS4vMDEyMzQ1Njc4OTo7PD0+P0BBQkNERUZHSElKS0xNTk9QUVJTVFVWV1hZWltcXV5fYGFiY2RlZmdoaWprbG1ub3BxcnN0dXZ3eHl6e3x9fn+AgYKDhIWGh4iJiouMjY6PkJGSk5SVlpeYmZqbnJ2en6ChoqOkpaanqKmqq6ytrq+wsbKztLW2t7i5uru8vb6/wMHCw8TFxsfIycrLzM3Oz9DR0tPU1dbX2Nna29zd3t/g4eLj5OXm5+jp6uvs7e7v8PHy8/T19vf4+fr7/P3+/w==",
+              },
+            ],
+            "location": {
+              "column": 9,
+              "file": "<root>/basic.test.ts",
+              "line": 12,
+            },
+            "type": "with Uint8Array body",
+          },
+          {
+            "attachments": [
+              {
+                "body": "",
+                "contentType": "text/plain",
+              },
+            ],
+            "location": {
+              "column": 9,
+              "file": "<root>/basic.test.ts",
+              "line": 13,
+            },
+            "type": "with contentType",
+          },
+        ],
+      }
+    `)
+    expect(annotations).toMatchInlineSnapshot(`
+      {
+        "second": [],
+        "simple": [],
+      }
+    `)
+  })
+
+  test('cannot annotate tests when the test finished running', async () => {
+    const { stderr } = await runInlineTests({
+      'basic.test.ts': `
+        import { test, recordArtifact } from 'vitest'
+        test('finished early', ({ task }) => {
```

Comment:
The annotations use this test with `globals: true` and no imports, but if I enable that `task` is `undefined`. Is this the expected behavior or is it a bug?

Reply:
Weird, it should always be defined